src/client/ua_client.c

```diff
@@ -291,17 +295,26 @@ static UA_StatusCode SecureChannelHandshake(UA_Client *client, UA_Boolean renew)
 
     retval = response.responseHeader.serviceResult;
     if(retval == UA_STATUSCODE_GOOD) {
+        /* Does the sequence number match? */
+        retval = UA_SecureChannel_checkSequenceNumber(seqHeader.sequenceNumber, client->channel);
+        if (retval != UA_STATUSCODE_GOOD){
+            UA_LOG_INFO_CHANNEL(client->config.logger, client->channel,
+                                "The sequence number was not increased by one. Got %i, expected %i",
+                                seqHeader.sequenceNumber, client->channel->receiveSequenceNumber + 1);
+            return UA_STATUSCODE_BADINTERNALERROR;
```

Comment:
Big memleak here.
Please clean up before returning (same as in line 290).
And check commits for issues with clang analyzer.